doc/examples/nn_deformable_registration_2d.py

```diff
@@ -0,0 +1,272 @@
+
+"""
+Create the data loader objects for the training, testing and the 
+sample sets.
+"""
+
+train_loader = DataLoader(x_train, static, batch_size=batch_size, shuffle=True)
+test_loader = DataLoader(x_test, static, batch_size=batch_size, shuffle=True)
+sample_loader = DataLoader(x_sample, static, shuffle=True)
+
+"""
+After completing the data processing, we will instantiate and compile our 
+deep learning model with the loss and the optimizer. In this example, 
+we will use the normalized cross-correlation (NCC) loss and the stochastic 
+gradient descent (SGD) as the optimizer. We also need to specify the input 
+shape in the form of (height, width, channels). Here, it is equal (32, 32, 1).
+"""
+
+loss = normalized_cross_correlation_loss()
+optimizer = tf.keras.optimizers.SGD(learning_rate=lr)
+
+# Compile the model with the loss and the optimizer.
+model = FCN2d(input_shape=(32, 32, 1), optimizer=optimizer, loss=loss)
+# model.compile(loss=loss, optimizer=optimizer)  # this works too
+
+"""
+Now that everything is set up, we can go ahead and train our model by 
+passing the training and testing data loaders to the ``fit`` method along 
+with the number of epochs we want to train the model for. After the 
+optimization is done, a history object is returned which records the 
+training and validation losses for every epoch.
+"""
+
+hist = model.fit(train_loader, epochs=epochs, validation_data=test_loader)
+
+"""
+Let's plot the losses to see if the model converged. 
+"""
+
+plt.plot(hist.history['loss'], color='royalblue', label='Train')
+plt.plot(hist.history['val_loss'], color='seagreen', label='Test')
+plt.xlabel('Epochs')
+plt.ylabel('Loss')
+plt.legend(loc="upper right")
+plt.title('Loss vs. Epochs')
+plt.show()
+plt.savefig('loss_plot.png')
+
+"""
+.. figure:: loss_plot.png
+   :align: center
+
+    Training loss and testing loss w.r.t. epochs.   
+"""
+
+"""
+Now let's evaluate the trained model on the testing and sample sets. And 
+also compute the average NCC values for these sets. The NCC is just the 
+negative of the NCC loss. The NCC values range from -1 to 1. For a perfect 
+match, it is equal to 1 and for complete dissimilarity, it is -1. 
+"""
+
+test_loss = model.evaluate(test_loader)
+sample_loss = model.evaluate(sample_loader)
+
+test_ncc = -test_loss
+sample_ncc = -sample_loss
+
+print('Test NCC: ', test_ncc)
+print('Sample NCC: ', sample_ncc)
+
+"""
+Now let's warp the sample images and see if they get similar to the static 
+image. The returned registered images by the model are currently in the [0, 
+1] range. So we need to perform some post-processing to convert them to the 
+normal 8-bit images.
+"""
+
+moved = model.predict(sample_loader)  # shape (num_samples, 32, 32, 1)
+# Note: 'predict' returns outputs as numpy arrays.
+
+"""
+Finally, let's visualize the transformed image together with the static and 
+the moving images.
+"""
+
+moved = moved.squeeze(axis=-1)  # Remove the channel dim.
```

Comment:
This is only needed for single-channel images. Since most images are single-channel I also think it is useful to implement as library code. Do you mean something like this? Add the squeeze option in the predict method

```python
moved = model.predict(data, squeeze=True)
```